### PR TITLE
Update requirements.txt

### PR DESCRIPTION
**Pull request by jmcarpenter2**
### Description

I understand that this code is never ran, but my company uses ECR image scanning to identify vulnerabilities and are required to resolve any high or critical severity vulnerabilities within a short period for compliance. As this is in conda, any images containing conda have this vulnerability listed in the image scan. In my case, that is the jupyter/scipy-notebook. Hoping to resolve the issue completely by simply updating the requirement for the test.

https://nvd.nist.gov/vuln/detail/CVE-2023-30861

Switching away from updating flask regularly to using a conda-maintained package with no known security vulnerabilities.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~
    - As this change doesn't impact functionality,  I am not updating documentation

### CODE REVIEW

**Comment by jmcarpenter2** [comment]
@conda-bot  check

**Comment by jmcarpenter2** [comment]
I have signed the conda CLA now.

**Comment by jmcarpenter2** [comment]
@conda-bot check

**Comment by dholth** [comment]
Maybe we can find a fake package to use that will never earn cves. Ken were
you using one recently in a test?

**Comment by jezdez** [comment]
@conda-bot check

File: tests/conda_env/support/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-flask==1.1.1
+flask==2.3.3
```
Comment (jezdez) [inline tests/conda_env/support/requirements.txt:1]: Let's use a package that we have control over and that is unlikely to receive a CVE, or at least then it's on us to fix this test.

```suggestion
conda-package-handling==2.2.0
```

This'll also need an update of line https://github.com/jmcarpenter2/conda/blob/32e4661243cf863f17b71bad763dfc2e6630bf02/tests/conda_env/specs/test_requirements.py#L27

**Review by jezdez — CHANGES_REQUESTED** [review]
Let's change the package that we test for to one of ours and also please add a news file.

File: tests/conda_env/support/requirements.txt (hunk 1 above)
Comment (dholth) [reply tests/conda_env/support/requirements.txt:1]: Do those tests even check that the package is real or exists in the index?

Looks like a good choice as long as the environment under test wouldn't automatically get conda-package-handling as a conda dependency.

Comment (jezdez) [reply tests/conda_env/support/requirements.txt:1]: Only `test_environment` actually tests the parsing, the others only if the file is there and valid, which could be any matchspec compatible string IIUC

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Review by jezdez — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Comment by jmcarpenter2** [comment]
Ah great, thanks so much!

**Pull request merged**